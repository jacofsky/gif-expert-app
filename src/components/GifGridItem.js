import React from 'react'
import PropTypes from 'prop-types'

const GifGridItem = ({id, img, title, username}) => {
    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={img} alt={title} />
            <h3>{title}</h3>
            <p>By: {username ? username : 'Sin nombre'}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired
}

export default GifGridItem
