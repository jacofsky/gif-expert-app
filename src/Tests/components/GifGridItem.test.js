import '@testing-library/jest-dom';
import react from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem'

describe('Test in GifGridItem.test.js', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg'
    const username = 'NN'
    const wrapper = shallow(<GifGridItem title={title} img={url} username={username}/>)

    test('should show correctly the component', () => {

        expect(wrapper).toMatchSnapshot()
    })

    test('should title be in a h3 and alt', () => {

        const h3 = wrapper.find('h3')
        expect(h3.text().trim()).toBe(title)
        
    })

    test('should img equal url and alt props', () => {

        const img = wrapper.find('img')
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)

        
    })

    test('should have animated class', () => {

        const div = wrapper.find('div')
        const className = div.prop('className')

        expect(className.includes('animate__fadeIn')).toBe(true)

    })  
    
})
