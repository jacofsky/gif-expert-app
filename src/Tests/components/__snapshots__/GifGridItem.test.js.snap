// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifGridItem.test.js should show correctly the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <h3>
    Un titulo
  </h3>
  <p>
    By: 
    NN
  </p>
</div>
`;
